### ISSUE TITLE
NHM calls ETH/Pascal miner instead of ETH/SIA when switching.

### ISSUE THREAD

**Issue by xChackOx** [DillonN/NiceHashMiner#13]
Hi!, just made an account to report this:

I was observing the console, to learn more about NHM internals and something weird happened: Claymore's ETH/SIA was running, then NHM switched to Blake2s because it was more profitable. After a while ETH/SIA was profitable again, but NHM called Claymore's ETH/PASC instead of ETH/SIA and kept using the values of ETH/SIA after that, so it never switched back to ETH/SIA. I'm attaching the part of the log where this happened.

It happened once though, because I stopped NHM to force the switch and ETH/SIA kept running as the most profitable algo, so I couldn't reproduce the error.

[log.txt](https://github.com/DillonN/NiceHashMiner/files/1093323/log.txt)

**Comment by DillonN** [comment]
Thanks, I'll look into this

**Comment by DillonN** [comment]
This is interesting, I haven't been able to reproduce it and went over the code a bunch of times but couldn't find any problems. However from your log it definitely did happen

I made one small tweak (eefded1) that may fix the problem, will be in the next release. Keeping this open in case anyone gets it again.

**Comment by xChackOx** [comment]
Thanks for your hard work. I'll keep an eye on my log in case it happens again.

**Comment by aganorian** [comment]
Not directly related to this issue but here what is happening with me:
Version 1.7.6.1, 1080ti
The program wont switch to the most profitable algo, for example, it goes from blake2-->EWBF Equihash even if the latter is worse than DualMining.
I tested and daggerSIA gives me more coins than EWBF Equihash right now but NHM wont switch to it unless i disable the other algos

**Comment by xChackOx** [comment]
I would recommend to enable the console and monitor when NHM gets the algo current profit. It specifically says when it will switch and why, also why it doesn't switch.

If you can reproduce what you are describing then I would also recommend opening a new issue and attaching a log to help DillonN debug it.

**Comment by aganorian** [comment]
Problem is i cannot predict when it switches. Inst it something the logs record?

**Comment by xChackOx** [comment]
Yes, in the logs it should appear exactly what you're describing. It would
be ideal if you can catch it when it happens then close NHM and copy the
log to desktop. You can then upload it here. NHM's log gets big pretty fast
because it records a lot of stuff, so it is easier to locate the issue if
you can catch it when it happens (issue should be near the end of the log).
You can always just attach the log you have right now but I guess it will
be a bit harder to find where this happened.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Libre
de virus. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

_______________
Ing. Jorge Santos

**Comment by DillonN** [comment]
@aganorian if there is an error in the switching logic it should show up in the logs (you can find log.txt files in the logs folder)

It's also possible that the values you have entered for the speeds are not the same as your overall hashrates (e.g. dual mining is more power intensive, so may cause thermal throttling overtime). NHM relies on these values to properly switch profits
